### PR TITLE
feat(amf): Paging 3513 timer support

### PR DESCRIPTION
**Pull request by Kaleem-Wavelabs**
Signed-off-by: Kaleemullah Mohammed <kaleemullah.mohammed@wavelabs.ai>

Tested with the paging and T3513 timer support changes using UT

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by panyogesh** [comment]
Related to Zenhub task #8463

**Review by ulaskozat — APPROVED** [review]
LGTM. Take care of the nit and label as ready to merge when it is ready.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 1]
```diff
@@ -1111,34 +1113,45 @@ static int paging_t3513_handler(zloop_t* loop, int timer_id, void* arg) {
...
+        "T3513: timer has expired for ue id: %d with timer id: %d,"
+        "Sending Paging request again\n",
+        ue_id, timer_id);
+    /*
+     * Increment the retransmission counter
+     */
+    OAILOG_ERROR(
+        LOG_AMF_APP, "T3513: Incrementing retransmission_count to %d\n",
+        paging_ctx->paging_retx_count);
+
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp:1150]: I am wondering what is the convention to use ERROR or INFO, re-transmission is not an error at all.

File: lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp [hunk 2]
```diff
@@ -79,6 +81,18 @@ int amf_handle_service_request(
...
+
+    if ((paging_ctx) &&
+        (NAS5G_TIMER_INACTIVE_ID != paging_ctx->m5_paging_response_timer.id) &&
+        (0 != paging_ctx->m5_paging_response_timer.id)) {
+      amf_app_stop_timer(paging_ctx->m5_paging_response_timer.id);
+      paging_ctx->m5_paging_response_timer.id = NAS5G_TIMER_INACTIVE_ID;
+      paging_ctx->paging_retx_count           = 0;
+      // Fill the itti msg based on context info produced in amf core
+      OAILOG_INFO(LOG_AMF_APP, "T3513: After stopping PAGING Timer\n");
+    }
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp:94]: We may in the magma [doc](https://docs.magmacore.org/docs/contributing/contribute_conventions) describe better what falls in each log level. Here I would suggest at least DEBUG loglevel.

**Comment by ulaskozat** [comment]
@Kaleem-Wavelabs could you take care of @lionelgo's comments?

File: lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp (hunk 2 above)
Comment (Kaleem-Wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp:94]: Done the changes as suggested

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp (hunk 1 above)
Comment (Kaleem-Wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp:1150]: Done the changes as suggested

**Comment by Kaleem-Wavelabs** [comment]
> @Kaleem-Wavelabs could you take care of @lionelgo's comments?

Fixed the comments

**Comment by ulaskozat** [comment]
@Kaleem-Wavelabs there are a number of github action comments that are legit.

**Pull request merged**